### DISCUSSION TITLE
How to build GCM for Ubuntu/arm64?

### DISCUSSION THREAD

**Discussion by MarkCallow** [git-ecosystem/git-credential-manager#1217] [Q&A] [answered]
I need to get GCM working on Linux/arm64. I tried using https://github.com/git-ecosystem/git-credential-manager/blob/main/src/linux/Packaging.Linux/install-from-source.sh on my Ubuntu/arm64 system but it ended up building (?) and installing an x86-64 version. I've now cloned this repo and want to build it but I've been unable to find any build instructions. I'm completely new to dotnet.

How do I build and install GCM from the repo source? Or how else might I get an arm64 version?

**Comment by hickford** [comment, marked as answer]
>  how else might I get an arm64 version?

Easiest is to install the GCM .NET tool https://github.com/git-ecosystem/git-credential-manager/blob/release/docs/install.md#net-tool

    dotnet tool install -g git-credential-manager

> I'm completely new to dotnet

You'll need to first install the .NET SDK https://learn.microsoft.com/en-us/dotnet/core/install/linux-ubuntu

**Reply by MarkCallow** [reply]
Thank you @hickford. This worked. One pain point is  how to install the .NET SDK. This could easily be fixed with a link on https://github.com/git-ecosystem/git-credential-manager/blob/release/docs/install.md#net-tool to instructions. Luckily I found the SDK was already installed, most likely from my having previously run `install-from-source.sh`, so the above command alone did the trick. I also had to remove the broken binaries installed by the script from `/usr/local/{bin.share/gcm}`

**Reply by ldennington** [reply]
Thanks for the suggestion - I've updated the docs with a link to .NET's install instructions. Note that, due to installation instructions differing based on the distro and version users are running, the best we are able to do is link to the page that provides further links for users to follow to use their respective package managers to install.

**Comment by hickford** [comment]
Alternatively, depending on your use case, you might try [git-credential-oauth](https://repology.org/project/git-credential-oauth) which is included in Ubuntu 23.04 and later https://repology.org/project/git-credential-oauth

    sudo apt install git-credential-oauth

**Reply by MarkCallow** [reply]
Unfortunately I  have to keep using Ubuntu 22 for a while longer.

**Comment by ldennington** [comment]
Unfortunately, we do not yet officially ship for Linux arm64, although that work is slated to coincide with Git's 2.42 release in September and is being tracked with [this issue](https://github.com/git-ecosystem/git-credential-manager/issues/606).

**Reply by MarkCallow** [reply]
Good to hear official support is in progress.
